pycontrails/core/vector.py

```diff
@@ -1010,6 +1064,35 @@ def create_empty(
         """
         return cls(data=_empty_vector_dict(keys or set()), attrs=attrs, copy=False, **attrs_kwargs)
 
+    @classmethod
+    def from_dict(
+        cls: Type[VectorDatasetType], obj: dict[str, Any], copy: bool = True
+    ) -> VectorDatasetType:
+        """Create instance from dict representation containing data and attrs.
+
+        Parameters
+        ----------
+        obj : dict[str, Any]
+            Dict representation of VectorDataset (e.g. `Vector.to_dict()`)
+        copy : bool, optional
+            Passed to VectorDataset constructor.
+            Defaults to True.
+
+        Returns
+        -------
+        VectorDatasetType
+            VectorDataset instance.
+        """
+        data = {}
+        attrs = {}
+        for k, v in obj.items():
+            if isinstance(v, (list | np.ndarray)):
```

Comment:
I think you need to change `isinstance(v, (list | np.ndarray))` to `isinstance(v, (list, np.ndarray))` for python 3.9. This works in python >= 3.10.

Reply:
whoops, thanks!

Reply:
4ac728fe8598e0001cb97e7a80f07abb4e14a2cc